components/script/dom/serviceworkerregistration.rs

```diff
@@ -52,14 +55,18 @@ impl ServiceWorkerRegistration {
         let active_worker =
             ServiceWorker::install_serviceworker(global, script_url.clone(), scope.clone(), true);
         active_worker.set_transition_state(ServiceWorkerState::Installed);
-        reflect_dom_object(
-            Box::new(ServiceWorkerRegistration::new_inherited(
-                &*active_worker,
-                scope,
-            )),
-            global,
-            Wrap,
-        )
+
+        let registration = ServiceWorkerRegistration::new_inherited(&*active_worker, scope);
+
+        registration
+            .navigation_preload
+            .set(Some(&NavigationPreloadManager::new(&global, &registration)));
+
+        reflect_dom_object(Box::new(registration), global, Wrap)
+    }
+
+    pub fn active(&self) -> Option<&Dom<ServiceWorker>> {
```

Comment:
This should return `Option<&ServiceWorker>` via `self.active.as_ref().map(|active| &*active)`